src/main/kotlin/org/rust/ide/console/RsConsoleView.kt

```diff
@@ -0,0 +1,105 @@
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.wm.IdeFocusManager
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.cargo.project.model.cargoProjects
+import org.rust.lang.RsLanguage
+import org.rust.lang.core.psi.*
+import org.rust.openapiext.toPsiFile
+import javax.swing.JComponent
+
+class RsConsoleView(project: Project, title: String, private val consoleRunner: RsConsoleRunnerImpl)
+    : LanguageConsoleImpl(project, title, RsLanguage), ObservableConsoleView {
+
+    private lateinit var myExecuteActionHandler: RsConsoleExecuteActionHandler
+
+    private val myInitialized: ActionCallback = ActionCallback()
+    private val codeFragment: RsReplCodeFragment? = virtualFile.toPsiFile(project) as? RsReplCodeFragment
+
+    init {
+        val virtualFile = virtualFile
+        virtualFile.putUserData(RUST_CONSOLE_KEY, true)
+        // Mark editor as console one, to prevent autopopup completion
+        historyViewer.putUserData(ConsoleViewUtil.EDITOR_IS_CONSOLE_HISTORY_VIEW, true)
+        super.setPrompt(">> ")
+        setUpdateFoldingsEnabled(false)
+    }
+
+    fun setExecutionHandler(consoleExecuteActionHandler: RsConsoleExecuteActionHandler) {
+        myExecuteActionHandler = consoleExecuteActionHandler
+    }
+
+    override fun requestFocus() {
+        myInitialized.doWhenDone {
+            IdeFocusManager.getGlobalInstance().requestFocus(consoleEditor.contentComponent, true)
+        }
+    }
+
+    override fun createCenterComponent(): JComponent {
+        // workaround for extra lines appearing in the console
+        val centerComponent = super.createCenterComponent()
+        historyViewer.settings.additionalLinesCount = 0
+        historyViewer.settings.isUseSoftWraps = false
+        consoleEditor.gutterComponentEx.background = consoleEditor.backgroundColor
+        consoleEditor.gutterComponentEx.revalidate()
+        consoleEditor.colorsScheme.setColor(EditorColors.GUTTER_BACKGROUND, consoleEditor.backgroundColor)
+
+        return centerComponent
+    }
+
+    fun print(text: String, attributes: Key<*>) {
+        print(text, outputTypeForAttributes(attributes))
+    }
+
+    private fun outputTypeForAttributes(attributes: Key<*>): ConsoleViewContentType {
+        return when {
+            attributes === ProcessOutputTypes.STDERR -> ConsoleViewContentType.ERROR_OUTPUT
+            attributes === ProcessOutputTypes.SYSTEM -> ConsoleViewContentType.SYSTEM_OUTPUT
+            else -> ConsoleViewContentType.getConsoleViewType(attributes)
+        }
+    }
+
+    fun initialized() {
+        myInitialized.setDone()
+    }
+
+    fun updateCodeFragmentContext() {
+        codeFragment ?: return
+
+        val allCommandsText = consoleRunner.commandHistory.getAllCommandsText()
+        codeFragment.context = createContext(project, codeFragment.crateRoot as RsFile?, allCommandsText)
+    }
+
+    companion object {
+        private var RUST_CONSOLE_KEY: Key<Boolean> = Key("RS_CONSOLE_KEY")
```

Comment:
`var` is redundant because `Key` is mutable itself, and you don't need to reassign the field